### PR TITLE
Fix dependency constraints (Closes #1200)

### LINKED ISSUE #1200: Update dependency constraints for NumPy 2.0.0
Thanks for all of your work (especially on NumPy 2 compatibility)! 🙏

Was just noticing that packages are currently constrained to installing on NumPy 1

Think we would want to relax this to `>= 1.20.0`

https://github.com/PyTables/PyTables/blob/13ee9ac0dfe231b1e96ab3c1c53ed2c6dfd5ad20/pyproject.toml#L70-L71

Also would suggest using NumPy 2 during the build. So changing 1.20.0 to 2.0. The resulting binaries will be compatible with NumPy 1 & 2 (even though they are built with NumPy 2)

https://github.com/PyTables/PyTables/blob/13ee9ac0dfe231b1e96ab3c1c53ed2c6dfd5ad20/pyproject.toml#L1-L8

Sorry to bring this up right after a release 😞

### PR DESCRIPTION
**Pull request by avalentino**

### CODE REVIEW

**Comment by avalentino** [comment]
@jakirkham please feel free to review

**Review by FrancescAlted — APPROVED** [review]
Is the c-blosc submodule update desired in this PR?  Other than that, this looks good to me.

**Review by ivilata — APPROVED** [review]
Same comment as @FrancescAlted regarding `c-blosc`, otherwise LGTM. Thanks @avalentino!

**Comment by avalentino** [comment]
I have assigned #1197 (conversion of the c-blosc directory into a git sub-module) to the milestone 3.10.1, but I would prefer to address it as a separate PR.

Meanwhile I see that we have a test failure (segmentation fault) with numpy 2.1.dev0.
The problem seems to be in `test_write_chunk_missing1.
Any help on this would be welcome.

**Comment by avalentino** [comment]
>> Is the c-blosc submodule update desired in this PR?
> I have assigned https://github.com/PyTables/PyTables/issues/1197 (conversion of the c-blosc directory into a git sub-module) to the milestone 3.10.1, but I would prefer to address it as a separate PR.

Sorry for making confusion, #1197 is about HDF5-Blosc.
The c-blosc sub-module has been updated in #1193. Do we need another update?

**Comment by avalentino** [comment]
>>> Is the c-blosc submodule update desired in this PR?
>>
>> I have assigned https://github.com/PyTables/PyTables/issues/1197 (conversion of the c-blosc directory into a git sub-module) to the milestone 3.10.1, but I would prefer to address it as a separate PR.
>
> Sorry for making confusion, https://github.com/PyTables/PyTables/issues/1197 is about HDF5-Blosc.
> The c-blosc sub-module has been updated in https://github.com/PyTables/PyTables/pull/1193. Do we need another update?

OK sorry for the noise
It should be fixed now

File: pyproject.toml [hunk 1]
```diff
@@ -5,7 +5,10 @@
 requires = [
     "setuptools >= 61.0.0",
     "wheel",
-    "numpy >= 1.20.0",
+    # NOTE: it is still possible to build PyTables with numpy 1.x
+    # Please refer to https://github.com/PyTables/PyTables/issues/1200
+    # "numpy >= 1.20.0",
+    "numpy >= 2",
```
Comment (jakirkham) [inline pyproject.toml:11]:
```suggestion
    # NOTE: it is still possible to build PyTables with numpy 1.x
    # Please refer to https://github.com/PyTables/PyTables/issues/1200
    # If you change this to build with `numpy` 1, please update `numpy` in `dependencies` as well
    # "numpy >= 1.20.0,<2",
    "numpy >= 2",
```

File: pyproject.toml [hunk 2]
```diff
@@ -68,7 +71,7 @@ classifiers = [
 # ``requirements.txt``,
 # `jobs.build.steps` in `.github/workflows/ubuntu.yml`.
 dependencies = [
-    "numpy >= 1.20.0, <2",
+    "numpy >= 1.20.0",
```
Comment (jakirkham) [inline pyproject.toml:74]:
```suggestion
    # When building with `numpy` version `1`, change this constraint to `numpy` 1 only
    # "numpy >= 1.20.0,<2",
    "numpy >= 1.20.0",
```

**Review by jakirkham — APPROVED** [review]
Thanks Antonio! 🙏

Had some minor suggestions on the comments to provide a bit more guidance to users building for NumPy 1 specifically. Though no strong feelings on this or other wording

The requirements look good

**Comment by avalentino** [comment]
OK The test failure seems to be unrelated.
Merging

**Pull request merged**